### PR TITLE
Code switch

### PR DESCRIPTION
**Pull request by sgugger**
**Needs to be merged after the corresponding PR on moon-landing is merged and with an update in the Transformers doc**

This PR uses the new syntax for code-blocks when converting old rst files, or when converting new MDX files to notebooks. It shouldn't be merged before that new syntax is expected on the front side, and the Transformers doc is updated. I'll do the latter once the PR on the front side is closed to be merged, as it's an easy search and replace, and those docs files may be adapted as Steven begins to rewrite the docs.

### CODE REVIEW

**Review by mishig25 — COMMENTED** [review]
lgtm, working on the front side of it rn 👍

File: src/doc_builder/convert_rst_to_mdx.py [hunk 1]
```diff
@@ -550,8 +550,11 @@ def split_pt_tf_code_blocks(text):
                     code_lines["common"].append(lines[idx])
                 idx += 1
             if len(code_lines["pytorch"]) > 0 or len(code_lines["tensorflow"]) > 0:
-                block_lines = code_lines["common"].copy() + code_lines["pytorch"]
-                block_lines += ["===PT-TF-SPLIT==="] + code_lines["tensorflow"] + ["```"]
+                block_lines = ["<frameworkcontent>", "<pytorch>"]
```
[the remark this answers is not in the document]
Comment (sgugger) [reply src/doc_builder/convert_rst_to_mdx.py:553]: In that case we should use tf for TensorFlow, but I'm fine with that.

[the remark this answers is not in the document]
Comment (mishig25) [reply src/doc_builder/convert_rst_to_mdx.py:553]: yes, that works for me as well

**Review by LysandreJik — APPROVED** [review]
That works for me :)

**Review by mishig25 — APPROVED** [review]
lgtm!

**Pull request merged**